src/app/components/secret-key/mnemonic-key/mnemonic-input-field.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useState } from 'react';
+
+import { TextField } from '@radix-ui/themes';
+import { useField } from 'formik';
+import { css } from 'leather-styles/css';
+import { FlexProps, styled } from 'leather-styles/jsx';
+
+import { useIsFieldDirty } from '@app/common/form-utils';
+
+interface InputFieldProps extends FlexProps {
+  dataTestId?: string;
+  name: string;
+  value: string;
+  onPaste: (e: React.ClipboardEvent<HTMLInputElement>) => void;
+  onUpdateWord: (word: string) => void;
+  hasError?: boolean;
+  wordlist: string[];
+}
+export function InputField({ dataTestId, name, onPaste, onChange, value }: InputFieldProps) {
+  const [field, meta] = useField(name);
+  const [isFocused, setIsFocused] = useState(false);
+  const isDirty = useIsFieldDirty(name);
+
+  const getBorderColour = useCallback(() => {
+    if (isDirty && meta.error) {
+      // FIXME #4130: show a red border on error - couldn't get it working
+      return 'colors.error';
+    }
+    if (isFocused) {
+      return 'accent.text-primary';
+    }
+    return 'none';
+  }, [isDirty, isFocused, meta.error]);
+
+  return (
+    <TextField.Root
+      /**
+       * Focus styling:
+       * radix inserts a <div inside the root with .rt-TextFieldChrome
+       * onFocus - this adds an unwanted box-shadow
+       * setting color="brown" makes this transparent
+       * then focus border can be controlled more easily
+       */
+      color="brown"
+      className={css({
+        display: 'flex',
+        alignSelf: 'stretch',
+        height: '48px',
+        width: '176px',
+        gap: 'space.01',
+        px: 'space.03',
+        border: '2px solid',
+        borderColor: getBorderColour(),
+        borderRadius: '4px',
+      })}
```

Comment:
Not entirely sure how this works. Thought Panda doesn't work with runtime values?

Anyway, noticed that changing focus state with border, which moves the number when it's focused.

Common approach I've used to fix this is to apply temporary borders with pseudo elements ala
```
      className={css({
        position: 'relative',
        display: 'flex',
        alignSelf: 'stretch',
        height: '48px',
        width: '176px',
        gap: 'space.01',
        px: 'space.03',
        _focusWithin: {
          _before: {
            content: '""',
            zIndex: 999999,
            top: 0,
            right: 0,
            bottom: 0,
            left: 0,
            background: 'transparent',
            position: 'absolute',
            border: '2px solid',
            borderRadius: '4px',
          },
        },
      })}
```
note doesn't work with err state*

Reply:
By looks of it, I don't think it does change border colour as expected 
<img width="723" alt="image" src="https://github.com/leather-wallet/extension/assets/1618764/00400dea-e630-4efc-b3be-9b2f92597035">

Reply:
I noticed the same

Reply:
Yes it's not updating the colour properly. I'll take a look at what you suggested, thanks!

Reply:
I have always had to use `!important` to force the borderColor attribute to change, hope you can find something better here. It is the only prop I can't get to change.

Reply:
Thanks @fbwoolf . I think I tried `important` too but I will check where you've used it elsewhere. It was annoying to get this to work. 

It was fine setting the border to red on error just using a regular input but trickier with radix `TextField`

Reply:
I got this working using psuedo elements, thanks @kyranjamie - it's looking a lot better and less jumpy. I had to decrease the z-index value as otherwise sometimes I would have to double click on the field. 

Thanks  for helping too @fbwoolf. I think the way to update the border colour without using `!important` is to specify it directly on the `borderColor` property. I got this to work but I see now what you mean about sometimes needing it with panda.